islandora_compound_object.module

```diff
@@ -268,17 +268,94 @@ function islandora_compound_object_access(AbstractObject $object) {
 }
 
 /**
- * Replaces parent object thumbnail with child's thumbnail.
+ * Implements hook_compound_cmodel_islandora_derivative().
  */
-function islandora_compound_object_update_parent_thumbnail($parent) {
+function islandora_compound_object_compound_cmodel_islandora_derivative(AbstractObject $object = NULL, $ds_modified_params = array()) {
+  $module_path = drupal_get_path('module', 'islandora_solution_pack_compound');
+  return array(
+    array(
+      'source_dsid' => NULL,
+      'destination_dsid' => 'TN',
+      'function' => array('islandora_compound_object_update_parent_thumbnail'),
+      'file' => "$module_path/islandora_compound_object.module",
+    ),
+  );
+
+}
+
+/**
+ * Implements hook_islandora_derivative().
+ */
+function islandora_compound_object_islandora_derivative(AbstractObject $object) {
+  return array(
+    array(
+      'source_dsid' => 'TN',
+      'destination_dsid' => NULL,
+      'function' => array('islandora_compound_object_update_parent_thumbnail_from_child'),
+    ),
+  );
+}
+
+/**
+ * Update parent from child object.
+ *
+ * Given a child object, check to see if it is part of
+ * a compound and update the parent's TN datastream.
+ *
+ * @param AbstractObject $child
+ *   The potential child object.
+ */
+function islandora_compound_object_update_parent_thumbnail_from_child(AbstractObject $child) {
   module_load_include('inc', 'islandora', 'includes/utilities');
+  // Short circuit if feature is disabled.
+  if (!variable_get('islandora_compound_object_tn_deriv_hooks', FALSE)) {
```

Comment:
makes sense to me. I wanted to have it as soon as possible in the logic anyways. I just didn't think of putting it there.